### DISCUSSION TITLE
TMDB

### DISCUSSION THREAD

**Discussion by hirockstar** [j471n/j471n.in#18] [Q&A] [answered]
how to import movies please doc for that too.
also can we also import the fav cast too??

**Comment by j471n** [comment, marked as answer]
Hello Rock Star,

I am assuming you are asking this because of the **Recent movies** section in [about](https://j471n.in/about) page.
![image](https://user-images.githubusercontent.com/55713505/217004246-3a4f1527-9d1a-4044-bd06-247b207c50b6.png)

Actually, I am not using IMDB or it's API (because it's work on subscription model). You have to pay for it to use. You can get more info about in it's [documentation](https://developer.imdb.com/documentation/api-documentation/getting-access/).

I created a SQL database at [supabase](https://supabase.com/) which consists of the following columns-

- name
- rating
- watched
- image
- URL

From there I am fetching these details in my portfolio.

If you want to get the details through an API then you can use [TMDB](https://www.themoviedb.org/) and it's [API](https://www.themoviedb.org/documentation/api) for free.

**Reply by hirockstar** [reply]
oh yeah sorry i forgot that its TMDB BTW can you share the SQL file so i can import.
also how we can import movie to database??

**Reply by j471n** [reply]
Actually I can't share you a SQL file because Supabase doesn't allow to export the data. But what I can do is that I can share the data and with the help of this you will be able to create your own dataset. 

```json
[
    {
        "id": 9,
        "created_at": "2023-01-31T09:50:31.366171+00:00",
        "name": "The Last of Us",
        "image": "https://m.media-amazon.com/images/M/MV5BZGUzYTI3M2EtZmM0Yy00NGUyLWI4ODEtN2Q3ZGJlYzhhZjU3XkEyXkFqcGdeQXVyNTM0OTY1OQ@@._V1_FMjpg_UY720_.jpg",
        "url": "https://www.imdb.com/title/tt3581920",
        "year": 2023,
        "watched": false,
        "rating": null
    },
    {
        "id": 8,
        "created_at": "2023-01-30T05:28:00.819385+00:00",
        "name": "Pathaan",
        "image": "https://m.media-amazon.com/images/M/MV5BM2QzM2JiNTMtYjU4Ny00MDZkLTk3MmUtYTRjMzVkZGJlNmYyXkEyXkFqcGdeQXVyMTUzNTgzNzM0._V1_FMjpg_UX602_.jpg",
        "url": "https://www.imdb.com/title/tt12844910/",
        "year": 2023,
        "watched": true,
        "rating": 8
    },
]
```

I am also providing you some commands that will help you to create table.

```sql
/* Creating a table */
CREATE TABLE movies(
   id         INTEGER  NOT NULL PRIMARY KEY 
  ,created_at VARCHAR(32) NOT NULL
  ,name       VARCHAR(25) NOT NULL
  ,image      VARCHAR(138) NOT NULL
  ,url        VARCHAR(38) NOT NULL
  ,year       INTEGER  NOT NULL
  ,watched    VARCHAR(5) NOT NULL
  ,rating     INTEGER 
);

/* INSERT QUERY NO: 1 */
INSERT INTO movies(`id`, `created_at`, `name`, `image`, `url`, `year`, `watched`, `rating`)
VALUES (9, '2023-01-31T09:50:31.366171+00:00', 'The Last of Us', 'https://m.media-amazon.com/images/M/MV5BZGUzYTI3M2EtZmM0Yy00NGUyLWI4ODEtN2Q3ZGJlYzhhZjU3XkEyXkFqcGdeQXVyNTM0OTY1OQ@@._V1_FMjpg_UY720_.jpg', 'https://www.imdb.com/title/tt3581920', 2023, 'false', '');

/* INSERT QUERY NO: 2 */
INSERT INTO movies(`id`, `created_at`, `name`, `image`, `url`, `year`, `watched`, `rating`)
VALUES (8, '2023-01-30T05:28:00.819385+00:00', 'Pathaan', 'https://m.media-amazon.com/images/M/MV5BM2QzM2JiNTMtYjU4Ny00MDZkLTk3MmUtYTRjMzVkZGJlNmYyXkEyXkFqcGdeQXVyMTUzNTgzNzM0._V1_FMjpg_UX602_.jpg', 'https://www.imdb.com/title/tt12844910/', 2023, 'true', '8');
```

About importing movies to database, I do it manually in supabase. I might create an automated script in the future.

[movies.zip](https://github.com/j471n/j471n.in/files/10665331/movies.zip) contains the commands that I mentioned above.

**Reply by hirockstar** [reply]
got it bro keep it up thanks for your help.

**Reply by j471n** [reply]
Mention not bro. Feel free to ask anything.
